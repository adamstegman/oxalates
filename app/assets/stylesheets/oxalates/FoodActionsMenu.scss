@import "./actions-container";
@import "./FoodActionAdd.scss";
@import "./FoodActionDone.scss";
@import "./FoodActionEdit.scss";

.actions-container {
  @extend %actions-container;
}
