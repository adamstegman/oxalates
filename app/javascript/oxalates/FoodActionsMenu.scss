@import "./actions-container";

.actions-container {
  @extend %actions-container;
}
