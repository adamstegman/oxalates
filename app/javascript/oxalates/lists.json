[
  {
    "id": "all",
    "name": "All",
    "bottom_threshold": null,
    "top_threshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottom_threshold": 50,
    "top_threshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottom_threshold": 15,
    "top_threshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottom_threshold": 5,
    "top_threshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottom_threshold": null,
    "top_threshold": 5
  }
]
