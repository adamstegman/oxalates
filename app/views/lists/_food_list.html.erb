<ul class="food-list">
<% foods.each do |food| %>
  <li class="food-list-item <%= list_class_name(food.list) %>"><%= food.name %></li>
<% end %>
</ul>
