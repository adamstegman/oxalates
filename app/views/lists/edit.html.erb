<%= form_tag list_path(@list.id), method: :put do %>
  <ul class="food-list">
  <% @list.foods.each do |food| %>
    <li>
      <%= link_to food.name, edit_food_path(food), class: "edit-food-list-item #{list_class_name(food.list)}" %>
      <%= hidden_field_tag 'list[foods][][id]', food.id %>
      <%= label_tag 'list[foods][][name]', "Name", class: 'food-list-item-name' %>
      <%= text_field_tag 'list[foods][][name]', food.name, class: 'food-list-item-name' %>
    </li>
  <% end %>
  </ul>
<% end %>
