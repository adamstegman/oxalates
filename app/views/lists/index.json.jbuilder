json.lists(@lists) do |list|
  json.(list, :id, :name, :bottom_threshold, :top_threshold)
end
