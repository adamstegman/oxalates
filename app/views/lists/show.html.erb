<%= render "food_list", foods: @list.foods %>
