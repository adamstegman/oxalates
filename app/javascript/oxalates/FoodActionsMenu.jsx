import React from 'react';
import PropTypes from 'prop-types';

import { FoodActionAdd } from './FoodActionAdd';

import "./FoodActionsMenu.scss";

export class FoodActionsMenu extends React.Component {
  render() {
    const {
      activeListId,
      startNewFood,
    } = this.props;

    const onStartAddingFood = () => startNewFood(activeListId);
    let addButton = <FoodActionAdd onStartAddingFood={onStartAddingFood} />;

    return (
      <div className="actions-container">
        {addButton}
      </div>
    );
  }
};

FoodActionsMenu.propTypes = {
  activeListId: PropTypes.node,
  startNewFood: PropTypes.func.isRequired,
};
