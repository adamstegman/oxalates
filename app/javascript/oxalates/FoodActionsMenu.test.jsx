import React from 'react';
import renderer from 'react-test-renderer';

import lists from './__mocks__/lists.json';
import { FoodActionsMenu } from './FoodActionsMenu';

test('FoodActionsMenu renders food action buttons', () => {
  const component = renderer.create(
    <FoodActionsMenu activeListId={lists[0].id} startNewFood={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
