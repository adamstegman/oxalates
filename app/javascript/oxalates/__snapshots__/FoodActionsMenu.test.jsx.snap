// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodActionsMenu renders food action buttons 1`] = `
<div
  className="actions-container"
>
  <button
    className="food-action"
    onClick={[Function]}
    type="button"
  >
    Add
  </button>
</div>
`;
